src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,183 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Playwright.Transport
+{
+    internal class WebSocketTransport : IConnectionTransport, IDisposable
+    {
+        private const int DefaultBufferSize = 16000;  // Byte buffer size
+        private readonly ClientWebSocket _webSocket;
+        private readonly string _wsEndpoint;
+        private readonly CancellationTokenSource _readerCancellationSource;
+        private readonly CancellationTokenSource _connectCancellationSource;
+        private readonly BrowserTypeConnectOptions _options;
+        private readonly float _slowMo;
+        private readonly float _timeout;
+
+        internal WebSocketTransport(
+            string wsEndpoint = default,
+            BrowserTypeConnectOptions options = default)
+        {
+            _webSocket = new ClientWebSocket();
+            _wsEndpoint = wsEndpoint;
+            _options = options;
+            _slowMo = _options?.SlowMo ?? 0;
+            _timeout = _options?.Timeout ?? 30000;
+            _readerCancellationSource = new CancellationTokenSource();
+            _connectCancellationSource = new CancellationTokenSource((int)_timeout);
+            SetRequestHeaders();
+            _ = ConnectAsync();
```

Comment:
Move .ConnectAsync to the call site instead. You are in async there, so you can await `transport.ConnectAsync()` there and handle timeout in the right stack.

Reply:
Make sense, but was in thought process when websocket transport object is created it should start connecting. Cool will follow you approach where connect needs to be explicitly called.

Reply:
fixed